### PR TITLE
fix #5321 don't throw error when node gets relocated

### LINKED ISSUE #5321: Node relocation makes Vue throw error
### Version
2.2.6

### Reproduction link
[https://jsfiddle.net/rstoenescu/uqsath1t/2/](https://jsfiddle.net/rstoenescu/uqsath1t/2/)

### Steps to reproduce
Run jsfiddle, uncheck the tick box and look at browser console log. Vue tries to call Node.insertBefore() API but parent node does not match element's parent (which has been relocated in `mounted()` method). 

### What is expected?
Not to throw the error.

### What is actually happening?
Error is thrown: `DOMException: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.(…)`

---
Submitting fix in a bit. This helps with some features of Quasar Framework (Tooltip, Popover, and new ones currently unreleased).

### PR DESCRIPTION
**Pull request by rstoenescu**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
This PR fixes https://github.com/vuejs/vue/issues/5321
Reproduction link: https://jsfiddle.net/rstoenescu/uqsath1t/2/ (follow instructions in the Github issue)

### CODE REVIEW

**Comment by rstoenescu** [comment]
Referencing issue for this PR: https://github.com/vuejs/vue/issues/5321

File: src/core/vdom/patch.js [hunk 1]
```diff
@@ -237,7 +237,9 @@ export function createPatchFunction (backend) {
   function insert (parent, elm, ref) {
     if (isDef(parent)) {
       if (isDef(ref)) {
-        nodeOps.insertBefore(parent, elm, ref)
+        if (ref.parentNode && ref.parentNode === parent) {
```
Comment (yyx990803) [inline src/core/vdom/patch.js]: I think just `ref.parentNode === parent` is enough. `isDef(parent)` has already been checked.

**Comment by rstoenescu** [comment]
Redone with requested changes. Thanks!

**Comment by rstoenescu** [comment]
Thank you for merging!

**Pull request merged**